### PR TITLE
fix(performance): Do not set up filesystem on every call

### LINKED ISSUE #31357: Basic auth requests always set up the filesystem
ToDo
- [x] remove support for old chunking v1 in Desktop

If a request is authenticated and does not use cookies, it will always set up the filesystem.
The reason is the post-login gc we do:
https://github.com/nextcloud/server/blob/e6d9ef2e38daffcab808eaa41b18ab16c6253b97/lib/base.php#L815-L831

I think it is time to move that to a proper cron job instead. It would potentially clean up chunked upload attempts but there is no need to do this with every successful login?!

### PR DESCRIPTION
**Pull request by miaulalala**
Removed old OO_FileChunking logic that produced GC- collectable chunks

Added BG Job that will run a garbage collection for every user

Signed-off-by: Anna Larch <anna@nextcloud.com>

* Resolves: https://github.com/nextcloud/server/issues/31357
## Summary

Comparison for OCS requests:
https://blackfire.io/profiles/compare/2c3c3699-137e-49a3-a363-e0950bb278d4...41a33cf0-71ee-42aa-9791-9ef74f99e6c3/graph?settings%5Bdimension%5D=wt&settings%5Bdisplay%5D=landscape&settings%5BtabPane%5D=nodes&selected=&callname=main()&constraintDoc=
Screenshot 2023-02-16 at 16 52 52

## TODO

- [ ] Testing

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [x] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [ ] ~~Screenshots before/after for front-end changes~~ 
- [ ] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [ ] ~~[Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)~~

### CODE REVIEW

**Review by icewind1991 — DISMISSED** [review]

**Review by come-nc — APPROVED** [review]

File: lib/private/Server.php [hunk 1]
```diff
@@ -702,6 +702,8 @@ public function __construct($webRoot, \OC\Config $config) {
 		$this->registerDeprecatedAlias('CategoryFetcher', CategoryFetcher::class);
 
 		$this->registerService(ICache::class, function ($c) {
+			$logger = $c->get(LoggerInterface::class);
+			$logger->debug('File Cache is deprecated, please use the ICacheFactory instead. File Cache will be removed in the future');
```
Comment (come-nc) [inline lib/private/Server.php:706]: So maybe add a `@deprecated` to the class?
And a version so that we can safely remove it later.

Comment (ChristophWurst) [reply lib/private/Server.php:706]: And drop the log. Deprecation notices in the logfile are not necessary.

Comment (miaulalala) [reply lib/private/Server.php:706]: > And drop the log. Deprecation notices in the logfile are not necessary.

Was requested

File: lib/base.php [hunk 2]
```diff
@@ -855,23 +855,6 @@ public static function registerCleanupHooks(\OC\SystemConfig $systemConfig): voi
 					$throttler = Server::get(\OC\Security\Bruteforce\Throttler::class);
 					$throttler->resetDelay($request->getRemoteAddress(), 'login', ['user' => $uid]);
 				}
-
-				try {
-					$cache = new \OC\Cache\File();
-					$cache->gc();
```
[the remark this answers is not in the document]
Comment (miaulalala) [reply lib/base.php:861]: Added a BG job

File: apps/dav/lib/Connector/Sabre/Directory.php [hunk 3]
```diff
@@ -153,12 +140,8 @@ public function createFile($name, $data = null) {
 			$this->fileView->unlockFile($path . '.upload.part', ILockingProvider::LOCK_EXCLUSIVE);
 			$node->releaseLock(ILockingProvider::LOCK_SHARED);
 			return $result;
-		} catch (\OCP\Files\StorageNotAvailableException $e) {
-			throw new \Sabre\DAV\Exception\ServiceUnavailable($e->getMessage(), $e->getCode(), $e);
 		} catch (InvalidPathException $ex) {
 			throw new InvalidPath($ex->getMessage(), false, $ex);
-		} catch (ForbiddenException $ex) {
-			throw new Forbidden($ex->getMessage(), $ex->getRetry(), $ex);
```
[the remark this answers is not in the document]
Comment (miaulalala) [reply apps/dav/lib/Connector/Sabre/Directory.php:161]: Added back

File: apps/dav/lib/Connector/Sabre/Directory.php [hunk 4]
```diff
@@ -153,12 +140,8 @@ public function createFile($name, $data = null) {
 			$this->fileView->unlockFile($path . '.upload.part', ILockingProvider::LOCK_EXCLUSIVE);
 			$node->releaseLock(ILockingProvider::LOCK_SHARED);
 			return $result;
-		} catch (\OCP\Files\StorageNotAvailableException $e) {
-			throw new \Sabre\DAV\Exception\ServiceUnavailable($e->getMessage(), $e->getCode(), $e);
```
[the remark this answers is not in the document]
Comment (miaulalala) [reply apps/dav/lib/Connector/Sabre/Directory.php:157]: Added back

File: lib/private/Server.php (hunk 1 above)
Comment (ChristophWurst) [reply lib/private/Server.php]: By? This has a penalty on performance and bloats logfiles on dev envs.

**Review by come-nc — APPROVED** [review]

**Review by juliusknorr — APPROVED** [review]

**Comment by juliushaertl** [comment]
Last commit seems to indeed break subfolder listing of mount points :/ Needs another dive into why Depth > 0 requests for dav fail with shares being moved into those subfolders without that then.

**Comment by juliushaertl** [comment]
I finally figured out why the failure was happening. When using the nodes api in dav we initialise it with the wrong view fake root `/user/files` instead of ``. This doesn't cause issues as dav rarely calls nodes api endpoints that then access the view. Pushed a fix in this PR to reuse already passed LazyFolder instances. The full fix is separate in #36774

**Comment by DaphneMuller** [comment]
hello @miaulalala 
Thank you so much for your work on this pull request! This ticket seems to have the tag 'pending documentation'. 
Is there any chance you could clarify what documentation is missing? Is this for admins or for app developers?

**Comment by miaulalala** [comment]
This PR was reverted @DaphneMuller

**Pull request merged**